lib/io/sqlite/delete.js

```diff
@@ -0,0 +1,23 @@
+import { loadDB } from "./setup.js";
+import { Note } from "./models.js";
+import { deleteNoteContent, executeIfDefined } from "../util.js";
+
+function cascadeDeleteNote(note, callback, onFailure) {
+  loadDB()
+    .then(() => {
+      Note.forge({ id: note.id })
+        .destroy()
+        .then(() => {
```

Comment:
Using `await` might be better here. For example, you can use:

``` js
.then(async () => {
  await Note.forge({ id: note.id }).destroy().catch(err => { 
    throw new Error("Failed to load database: " + err);
  });
  deleteNoteContent(note);
  executeIfDefined(callback);
});
```

Reply:
cool. lessens the amount of closure nesting.